components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/grant/CibaGrantHandler.java

```diff
@@ -0,0 +1,372 @@
+                }
+            }
+        }
+        if (auth_req_id == null) {
+            // Authentication Request ID is missing.
+
+            if (log.isDebugEnabled()) {
+                log.debug("token request  misses mandated parameter (auth_req_id).");
+            }
+            throw new IdentityOAuth2Exception(MISSING_AUTH_REQ_ID);
+        }
+        if (!tokenReq.getGrantType().equals(CibaParams.OAUTH_CIBA_GRANT_TYPE)) {
+            // Grant Type is not expected to be for ciba.
+
+            throw new IdentityOAuth2Exception(INVALID_GRANT);
+        }
+        try {
+            SignedJWT signedJWT = SignedJWT.parse(auth_req_id);
+            JSONObject jo = signedJWT.getJWTClaimsSet().toJSONObject();
+            String authCodeDOKey = this.getCibaAuthCodeDOKeyFromAuthReqCodeHash(auth_req_id);
+
+            // Retrieving information from database and assign to CibaAuthCodeDO.
+            CibaAuthCodeDO cibaAuthCodeDO =
+                    CibaDAOFactory.getInstance().getCibaAuthMgtDAO().getCibaAuthCodeDO(authCodeDOKey);
+
+            // Validate polling for tokenRequest.
+            validatePolling(jo, auth_req_id, cibaAuthCodeDO);
+
+            this.setPropertiesForTokenGeneration(tokReqMsgCtx, cibaAuthCodeDO);
+            return true;
+        } catch (CibaCoreException e) {
+            throw new IdentityOAuth2Exception(INVALID_PARAMETERS);
+        } catch (ParseException e) {
+            throw new IdentityOAuth2Exception(INVALID_AUTH_REQ_ID);
+        }
+    }
+
+    /**
+     * Validates the polling for tokenRequest.
+     *
+     * @param auth_req_id    Authentication Request Identifier as JSON.
+     * @param authReqID      Authentication Request Identifier as String.
+     * @param cibaAuthCodeDO Persisted DO which accumulates authentication and token request information.
+     * @throws CibaCoreException       Exception thrown from CibaCore Component.
+     * @throws IdentityOAuth2Exception Exception thrown regarding IdentityOAuth
+     */
+    private void validatePolling(JSONObject auth_req_id, String authReqID, CibaAuthCodeDO cibaAuthCodeDO)
+            throws IdentityOAuth2Exception, CibaCoreException {
+
+        try {
+            // Validate whether provided authReqId is a valid.
+            validateAuthReqID(authReqID);
+
+            // Validate whether provided authReqId has a valid audience.
+            validateAudience(auth_req_id);
+
+            // Validate whether polling is allowed for the request made.
+            validatePollingAllowed(cibaAuthCodeDO);
+
+            // Validate whether provided authReqId is still active.
+            activeAuthreqID(cibaAuthCodeDO);
+
+            // Validate whether polling is under proper rate limiting.
+            validatePollingFrequency(cibaAuthCodeDO);
+
+            // Validate whether authentication  is provided with affirmative consent.
+            if (IsConsentGiven(cibaAuthCodeDO).equals(false)) {
```

Comment:
methods should be start with simple letters.
Instead of this -> (false).equals.(IsConsentGiven(cibaAuthCodeDO))